core/transaction/entity.go

```diff
@@ -249,77 +249,81 @@ func (t *Transaction) VerifySigWith(pubkey string, verifyHandler VerifyFunc) (bo
 }
 
 func SendTransactionSync(txn *Transaction, miners []string) error {
-	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
-	defer cancel()
+	const requestTimeout = 30 * time.Second // Timeout for each request
 
-	wg := sync.WaitGroup{}
-	wg.Add(len(miners))
 	fails := make(chan error, len(miners))
+	var wg sync.WaitGroup
 
 	for _, miner := range miners {
-		url := fmt.Sprintf("%v/%v", miner, TXN_SUBMIT_URL)
+		wg.Add(1)
+		minerURL := fmt.Sprintf("%v/%v", miner, TXN_SUBMIT_URL)
+
 		go func(url string) {
 			defer wg.Done()
-			select {
-			case <-ctx.Done():
```

Comment:
this logic was correct, you shouldn't revert it. we need to check if context is already cancelled/expired